bdk_chain/src/keychain/keychain_txout_index.rs

```diff
@@ -107,82 +134,144 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
+        &self.lookahead
+    }
+
+    /// Convenience method to call [`set_lookahead`] for all keychains.
+    ///
+    /// [`set_lookahead`]: Self::set_lookahead
+    pub fn set_lookahead_for_all(&mut self, lookahead: u32) {
+        for keychain in &self.keychains.keys().cloned().collect::<Vec<_>>() {
+            self.lookahead.insert(keychain.clone(), lookahead);
+            self.replenish_lookahead(&keychain);
+        }
+    }
+
+    /// Set the lookahead count for `keychain`.
+    ///
+    /// The lookahead is the number of scripts to cache ahead of the last stored script index. This
+    /// is useful during a scan via [`scan`] or [`scan_txout`].
+    ///
+    /// # Panics
+    ///
+    /// This will panic if `keychain` does not exist.
+    ///
+    /// [`scan`]: Self::scan
+    /// [`scan_txout`]: Self::scan_txout
+    pub fn set_lookahead(&mut self, keychain: &K, lookahead: u32) {
+        self.lookahead.insert(keychain.clone(), lookahead);
+        self.replenish_lookahead(keychain);
+    }
+
+    fn replenish_lookahead(&mut self, keychain: &K) {
+        let descriptor = self.keychains.get(keychain).expect("keychain must exist");
+        let next_index = self.last_revealed.get(keychain).map_or(0, |v| *v + 1);
+        let lookahead = self.lookahead.get(keychain).map_or(0, |v| *v);
+
+        for (new_index, new_spk) in range_descriptor_spks(
+            Cow::Borrowed(descriptor),
+            next_index..next_index + lookahead,
+        ) {
+            let _inserted = self
+                .inner
+                .insert_script_pubkey((keychain.clone(), new_index), new_spk);
+            debug_assert!(_inserted, "must not have existing spk");
+        }
     }
 
     /// Generates script pubkey iterators for every `keychain`. The iterators iterate over all
-    /// derivable scripts.
-    pub fn scripts_of_all_keychains(
-        &self,
-    ) -> BTreeMap<K, impl Iterator<Item = (u32, Script)> + Clone> {
+    /// derivable script pubkeys.
+    pub fn keychain_spks_of_all(&self) -> BTreeMap<K, impl Iterator<Item = (u32, Script)> + Clone> {
         self.keychains
             .iter()
             .map(|(keychain, descriptor)| {
                 (
                     keychain.clone(),
-                    descriptor_into_script_iter(descriptor.clone()),
+                    range_descriptor_spks(Cow::Owned(descriptor.clone()), 0..),
                 )
             })
             .collect()
     }
 
     /// Generates a script pubkey iterator for the given `keychain`'s descriptor (if exists). The
     /// iterator iterates over all derivable scripts of the keychain's descriptor.
-    pub fn scripts_of_keychain(
-        &self,
-        keychain: &K,
-    ) -> Option<impl Iterator<Item = (u32, Script)> + Clone> {
-        self.keychains
+    ///
+    /// # Panics
+    ///
+    /// This will panic if `keychain` does not exist.
+    pub fn keychain_spks(&self, keychain: &K) -> impl Iterator<Item = (u32, Script)> + Clone {
```

Comment:
same here. The old name was better but just replace scripts with spks.

Reply:
Fixed in d2b0d214e6c7819aa904237c967c46bdaf66ab5e